flask_httpauth.py

```diff
@@ -142,8 +154,18 @@ def authenticate(self, auth, stored_password):
             except TypeError:
                 client_password = self.hash_password_callback(username,
                                                               client_password)
-        return client_password is not None and \
-            client_password == stored_password
+        if stored_password is None or client_password is None:
+            return False
+
+        try:
+            stored_password = stored_password.encode("utf-8")
+        except AttributeError:
+            pass
+        try:
+            client_password = client_password.encode("utf-8")
+        except AttributeError:
+            pass
```

Comment:
We still need all of this stuff to ensure both passwords are the same type (the easiest way is to just convert both to bytes).

Reply:
Why would they be of different types? If they are different types then the current version of the package would report the passwords as non matching. Or am I missing something?